pkg/api/validation/validation.go

```diff
@@ -1519,7 +1532,11 @@ func ValidatePersistentVolume(pv *api.PersistentVolume) field.ErrorList {
 			allErrs = append(allErrs, field.Invalid(specPath.Child("storageClassName"), pv.Spec.StorageClassName, msg))
 		}
 	}
-
+	if pv.Spec.VolumeMode != nil && !utilfeature.DefaultFeatureGate.Enabled(features.BlockVolumeSupport) {
+		allErrs = append(allErrs, field.Forbidden(specPath.Child("volumeMode"), "PersistentVolume volumeMode is disabled by feature-gate"))
+	} else if pv.Spec.VolumeMode != nil && *pv.Spec.VolumeMode != api.PersistentVolumeFilesystem && *pv.Spec.VolumeMode != api.PersistentVolumeBlock {
```

Comment:
You can check the value of VolumeMode against supportedVolumeModes, instead of each explicit value.